src/main/java/org/broadinstitute/hellbender/utils/recalibration/covariates/ContextCovariate.java

```diff
@@ -188,7 +188,7 @@ private static IntList contextWith(final byte[] bases, final int contextSize, fi
             currentKey = 0;
             currentNPenalty = contextSize - 1;
             int offset = newBaseOffset;
-            while (bases[currentNPenalty] != 'N') {
+            while (BaseUtils.simpleBaseToBaseIndex(bases[currentNPenalty]) != -1) {
                 final int baseIndex = BaseUtils.simpleBaseToBaseIndex(bases[currentNPenalty]);
                 currentKey |= (baseIndex << offset);
                 offset -= 2;
```

Comment:
I reiterate my fear (expressed in person) that even if we make `ContextCovariate` able to handle (or at least not crash on) non-ACGT bases, there will be other parts of BQSR that can't handle them. We should wait until this branch has a successful run on the sample(s) that revealed this problem before merging.

Reply:
agreed. the production team has this branch and is running the failed sample against it.